appengine-push/test_deploy.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python
+# Copyright 2015 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""Test classes for the command line Cloud Pub/Sub sample."""
+
+import httplib
+import os
+import time
+import unittest
+import urllib
+import uuid
+
+
+TEST_GAE_HOST_ENV = 'TEST_GAE_HOST'
+DEFAULT_TEST_GAE_HOST = "py-dot-cloud-pubsub-sample-test.appspot.com"
+MAX_RETRY = 3
+SLEEP_TIME = 1
+
+
+def get_gae_host():
+    """Return the deployed application's endpoint used in the tests."""
+    return os.getenv(TEST_GAE_HOST_ENV, DEFAULT_TEST_GAE_HOST)
+
+
+class IntegrationTestCase(unittest.TestCase):
+    """A test case for the Pubsub App Engine sample."""
+
+    def setUp(self):
+        random_id = uuid.uuid4()
+        self.message = 'message-%s' % random_id
+        test_host = get_gae_host()
+        self.conn = httplib.HTTPSConnection(test_host)
+
+    def tearDown(self):
+        self.conn.close()
+
+    def test_get(self):
+        """Test accessing the top page."""
+        self.conn.request('GET', '/')
+        resp = self.conn.getresponse()
+        # This ensures that our App Engine service account is working
+        # correctly.
+        self.assertEquals(200, resp.status)
+
+    def fetch_messages(self):
+        """Fetch messages"""
+        conn = httplib.HTTPSConnection(get_gae_host())
```

Comment:
No strong readon, but httplib is in the standard lib and it satisfy my needs here.